### PR TITLE
Refactoring to reuse response status codes

### PR DESCRIPTION
**Pull request by Vendin**
Refactoring to reuse response status codes.
Status codes are needed, where we ourselves implement a cycle for all redirects.
To understand where the last redirect is.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -864,6 +859,14 @@ func getRedirectURL(baseURL string, location []byte) string {
 	return redirectURL
 }
 
+func IsFinishStatusCode(statusCode int) bool {
```
Comment (erikdubbelboer) [inline client.go]: Can you add some documentation to this function. Also a better name might be `StatusCodeIsRedirect`?

Comment (Vendin) [reply client.go]: Done. Check please.

File: client.go [hunk 2]
```diff
@@ -864,6 +859,17 @@ func getRedirectURL(baseURL string, location []byte) string {
 	return redirectURL
 }
 
+// StatusCodeIsRedirect checks the response code.
+//
+// If this is a redirect code then it returns true, otherwise false.
```
Comment (erikdubbelboer) [inline client.go]: Sorry but could you please change it to:
```
StatusCodeIsRedirect returns true if the status code indicates a redirect.
```

**Comment by erikdubbelboer** [comment]
Thanks!

**Pull request merged**